### PR TITLE
fix bootstrapper timestamps and resource details

### PR DESCRIPTION
**Pull request by andre-merzky**
Somewhere on the way, we lost the alignment of bootstrapper timestamps
in certain conditions.  This hotfix provides:

  - recording of timestamps independent of locale (LC_NUMERIC)
  - time alignment with `sync_abs` (previous `sync_rel`)

The PR also has a small fix which avoids overwriting of resource details
in some cases.

### CODE REVIEW

File: src/radical/pilot/agent/bootstrap_0.sh [hunk 1]
```diff
@@ -210,8 +213,20 @@ create_gtod()
 
     chmod 0755 ./gtod
 
-    TIME_ZERO=`./gtod`
-    export TIME_ZERO
+    # initialize profile
+    PROFILE="bootstrap_0.prof"
+    now=$(./gtod)
+    echo "#time,name,uid,state,event,msg" > "$PROFILE"
```
Comment (mtitov) [inline src/radical/pilot/agent/bootstrap_0.sh:219]: update as `#time,event,comp,thread,uid,state,msg`?

File: src/radical/pilot/agent/bootstrap_0.sh [hunk 2]
```diff
@@ -210,8 +213,20 @@ create_gtod()
...
+
+    ip=$(ip addr \
+         | grep 'state UP' -A2 \
+         | grep 'inet' \
+         | awk '{print $2}' \
+         | cut -f1 -d'/')
+    printf "%.4f,%s,%s,%s,%s,%s,%s\n" \
+        "$now" "sync_abs" "bootstrap_0" "MainThread" "$PILOT_ID" \
+        "PMGR_ACTIVE_PENDING" "$(hostname):$ip:$now:$now:$now" \
```
Comment (mtitov) [inline src/radical/pilot/agent/bootstrap_0.sh:228]: is this required to repeat `$now` 3 times?

**Review by mtitov — APPROVED** [review]
couple comments, but overall LGTM

File: src/radical/pilot/agent/bootstrap_0.sh (hunk 1 above)
Comment (andre-merzky) [reply src/radical/pilot/agent/bootstrap_0.sh:219]: Woah - good catch, that line must have quite a history of blind copy-and-paste... :-P

File: src/radical/pilot/agent/bootstrap_0.sh (hunk 2 above)
Comment (andre-merzky) [reply src/radical/pilot/agent/bootstrap_0.sh:228]: The `sync_abs` timestamps are normally derived from an NTP query, and for that we include the query send time, query receive time, and estimated time.  Here we basically fake a perfect NTP service - which is usually fine as we have second granularity only anyway.

**Pull request merged**